### ISSUE TITLE
Style/RegexpLiteral autocorrect breaks syntax with curly braces

### ISSUE THREAD

**Issue by positiv2** [rubocop/rubocop#15115]
### Before submitting

- [x] I have searched the existing issues and this hasn't been reported yet.
- [x] I am running the latest stable version of RuboCop.

### Issue type

Incorrect autocorrect

### Cop name

Style/RegexpLiteral

### Steps to reproduce the problem

1. Create a file with the following content
```rb
# frozen_string_literal: true

puts '{/'.match?(/{\//)
```
2. Run `rubocop -a (path to the file)`

### Expected behavior

Use a different separator instead of a curly brace after %r, or do not autocorrect.

### Actual behavior

Autocorrect introduces a syntax error. Using the file from "Steps to reproduce the problem":

Without autocorrect:
```
$ ~/tools/rubocop/exe/rubocop -c ./.rubocop.yml --debug regexp.rb
configuration from /home/positiv2/other/.rubocop.yml
Default configuration from /home/positiv2/tools/rubocop/config/default.yml
Use parallel by default.
Inspecting 1 file
Skipping parallel inspection: only a single file needs inspection
Scanning /home/positiv2/other/regexp.rb
C

Offenses:

regexp.rb:3:18: C: [Correctable] Style/RegexpLiteral: Use %r around regular expression.
puts '{/'.match?(/{\//)
                 ^^^^^

1 file inspected, 1 offense detected, 1 offense autocorrectable
Finished in 0.07372 seconds
```
```
$ ruby -c regexp.rb
Syntax OK
```

With autocorrect:
```
$ ~/tools/rubocop/exe/rubocop -c ./.rubocop.yml --debug regexp.rb -a
configuration from /home/positiv2/other/.rubocop.yml
Default configuration from /home/positiv2/tools/rubocop/config/default.yml
Use parallel by default.
Inspecting 1 file
Skipping parallel inspection: only a single file needs inspection
Scanning /home/positiv2/other/regexp.rb
F

Offenses:

regexp.rb:3:18: F: Lint/Syntax: unterminated regexp meets end of file; expected a closing delimiter
(Using Ruby 4.0 parser; configure using TargetRubyVersion parameter, under AllCops)
puts '{/'.match?(%r{{/})
                 ^^^
regexp.rb:3:18: C: [Corrected] Style/RegexpLiteral: Use %r around regular expression.
puts '{/'.match?(/{\//)
                 ^^^^^
regexp.rb:3:25: F: Lint/Syntax: unexpected end-of-input; expected a ) to close the arguments
(Using Ruby 4.0 parser; configure using TargetRubyVersion parameter, under AllCops)
puts '{/'.match?(%r{{/}) ...


1 file inspected, 3 offenses detected, 1 offense corrected
Finished in 0.07884 seconds
```
```
$ ruby -c regexp.rb
ruby: regexp.rb:3: syntax errors found (SyntaxError)
  1 | # frozen_string_literal: true
  2 |
> 3 | puts '{/'.match?(%r{{/})
    |                  ^~~ unterminated regexp meets end of file; expected a closing delimiter
> 4 |
    | ^ unexpected end-of-input; expected a `)` to close the arguments
```

### RuboCop configuration

```yaml
AllCops:
  TargetRubyVersion: 4.0
  NewCops: enable
```

### RuboCop version

```shell
$ ~/tools/rubocop/exe/rubocop -c ./.rubocop.yml -V
1.86.1 (using Parser 3.3.11.1, Prism 1.9.0, rubocop-ast 1.49.1, analyzing as Ruby 4.0, running on ruby 4.0.1) [x86_64-linux]
```
